lib/apipie/generator/swagger/type_extractor.rb

```diff
@@ -37,15 +37,16 @@ def extract_with_warnings(warnings = {})
   private
 
   def extract
-    expected_type = if string?
-                      :string
-                    elsif boolean?
-                      :boolean
-                    elsif enum?
-                      :enum
-                    else
-                      @validator.expected_type.to_sym
-                    end
+    expected_type =
+      if string?
+        :string
+      elsif boolean?
+        :boolean
+      elsif enum?
+        :enum
+      else
+        @validator.expected_type.to_sym
```

Comment:
now that I am reading this properly.. maybe this was the original code... but I wonder why the `if/elsif/elsif/else` when `@validator.expected_type.to_sym` probably already return the right value for `string`, `boolean`, and `enum` ?

Reply:
What was confusing to me as well.
There are a couple of issues we handle here 

`ParamDescription`'s validator can be 
 - `nil`
 - `Apipie::Validator::BaseValidator`
 -  `ResponseDescriptionAdapter::PropDesc::Validator`

Updated Yard comment dc88775d6adbbea6f038b83ae93abd2446207169

____

In case of nil we want to default to `string`

____

`Apipie::Validator::EnumValidator#expected_type` [returns `string`](https://github.com/PanosCodes/apipie-rails/blob/87584003dd7ca78821c79ccfa20c97520a201e40/lib/apipie/validator.rb#L80)
____

`ResponseDescriptionAdapter::PropDesc::Validator#expected_type` [can we whatever](https://github.com/PanosCodes/apipie-rails/blob/15584d7acc2bc7c477ef4403b6f0fa05a60c6732/lib/apipie/response_description_adapter.rb#L63) by if we give it values that are `[true, false]` we need to result into `boolean`.